templates/ntp.conf.j2

```diff
@@ -7,14 +7,20 @@ value['hostname'] }}{{
 ' maxpoll {0}'.format(value['maxpoll']) if 'maxpoll' in value else '' }}{{
 ' iburst' if 'iburst' in value and value else '' }}
 {% endfor %}
+{% for value in timesync_ntp_peers %}
+peer {{ value['hostname'] }}{{
+' minpoll {0}'.format(value['minpoll']) if 'minpoll' in value else '' }}{{
+' maxpoll {0}'.format(value['maxpoll']) if 'maxpoll' in value else '' }}{{
+' iburst' if 'iburst' in value and value else '' }}
+{% endfor %}
 
 driftfile /var/lib/ntp/drift
 
 # Permit time synchronization with our time source, but do not
 # permit the source to query or modify the service on this system.
-restrict default nomodify notrap nopeer noquery
+restrict default nomodify notrap {{ 'nopeer' if timesync_ntp_peers|length == 0 else '' }} noquery
 {% if timesync_ntp_version.stdout is version('4.2.6', '<') %}
-restrict -6 default nomodify notrap nopeer noquery
+restrict -6 default nomodify notrap {{ 'nopeer' if timesync_ntp_peers|length == 0 else '' }} noquery
 {% endif %}
```

Comment:
It's not clear to me why this is needed. As I understand it, `nopeer` applies only to unauthenticated packets when authentication is disabled with `disable auth`.

Reply:
I removed this in 17a3a7d6c112c2a686edf59372168b0121595143, I checked on my production and it was indeed wrong.